SQLTask/Task6.sql

```diff
@@ -0,0 +1,21 @@
+USE DB_Bank
+
+SELECT Clients.FullName,
+	AVG(TotalBalance.SumBalance) AS FullBalance,
+	IIF(SUM(Cards.CardBalance) IS NULL, 
+		0, 
+		SUM(Cards.CardBalance))
+	 AS CardsSum,
+	IIF(SUM(Cards.CardBalance) IS NULL, 
+		AVG(TotalBalance.SumBalance), 
+		AVG(TotalBalance.SumBalance) - SUM(Cards.CardBalance))
+	 AS AvailableMoney
+FROM (SELECT Clients.FullName, 
+		SUM(Accounts.Balance) AS SumBalance
+	FROM Clients
+	FULL OUTER JOIN Accounts ON Accounts.ClientId = Clients.Id
+	GROUP BY Clients.FullName) AS TotalBalance
+FULL OUTER JOIN Clients ON Clients.FullName = TotalBalance.FullName
+FULL OUTER JOIN Accounts ON Accounts.ClientId = Clients.Id
+FULL OUTER JOIN Cards ON Cards.AccountId = Accounts.Id
+GROUP BY Clients.FullName
```

Comment:
Always use some indexed value as GROUP BY column. Read more here   
https://stackoverflow.com/questions/70855216/what-will-be-faster-for-group-by-statement

Reply:
Resolve